padrino-core/lib/padrino-core/path_router/compiler.rb

```diff
@@ -34,12 +34,23 @@ def compiled?
         !!@compiled
       end
   
+
+      ##
+      # Finds routes by using request or env.
+      #
+      def find_by(request_or_env)
+        request = request_or_env.is_a?(Hash) ? Sinatra::Request.new(request_or_env) : request_or_env
+        pattern = encode_default_external(request.path_info)
+        verb    = request.request_method.downcase.to_sym
+        rotation { |offset| match?(offset, pattern) }.select { |route| route.verb == verb }
+      end
+
       ##
-      # Finds routes by using request.
+      # Calls routes by using request.
       #
-      def find_by_request(request)
+      def call_by_request(request)
         rotation do |offset|
-          pattern  = request.path_info.encode(Encoding.default_external)
+          pattern  = encode_default_external(request.path_info)
```

Comment:
Nope, it is on purpose.
The request instance must be able to modify to ensure a compatibility with sinatra.
See https://github.com/padrino/padrino-framework/blob/master/padrino-core/test/test_filters.rb#L390